### PR TITLE
initial tag support for replace_tags

### LINKED ISSUE #21: 12084.2 - implement tag replacement logic
Shippable/pm#12084

implement the logic behind reading the step.json and replacing the %% tags

### PR DESCRIPTION
**Pull request by trriplejay**
closes #21

### CODE REVIEW

File: _common/header.sh [hunk 1]
```diff
@@ -237,22 +237,54 @@ replace_tags() {
...
+    local token_list=($(grep "%%[A-Za-z0-9_\-\.]*%%" -o <<< $contents))
+    for token in "${token_list[@]}"; do
+      local json_path="${token//%%/}"
+      local json_result=$(_search_stepjson "$json_path")
+      if [ -z "$json_result" ]; then
+        echo "Error: match not found in step.json for pattern: $token. Skipping." >&2
+        continue
+      fi
+      local sed_result=$((echo $json_result|sed -r 's/([\$\.\*\/\[\\^])/\\\1/g'|sed 's/[]]/\[]]/g')>&1)
+      sed_token=$((echo $token|sed -r 's/([\$\.\*\/\[\\^])/\\\1/g'|sed 's/[]]/\[]]/g')>&1)
```
Comment (a-murphy) [inline _common/header.sh]: Is there a reason `sed_result` is `local` and `sed_token` is not?

File: _common/header.sh [hunk 2]
```diff
@@ -264,14 +296,77 @@ _shippable_replace_file() {
...
+    local token_list=($(grep "%%[A-Za-z0-9_\-\.]*%%" -o $temp_file))
+    for token in "${token_list[@]}"; do
+      local json_path="${token//%%/}"
+      local json_result=$(_search_stepjson "$json_path")
+      if [ -z "$json_result" ]; then
+        echo "Error: match not found in step.json for pattern: $token. Skipping." >&2
+        continue
+      fi
+      local sed_result=$((echo $json_result|sed -r 's/([\$\.\*\/\[\\^])/\\\1/g'|sed 's/[]]/\[]]/g')>&1)
+      sed_token=$((echo $token|sed -r 's/([\$\.\*\/\[\\^])/\\\1/g'|sed 's/[]]/\[]]/g')>&1)
```
Comment (a-murphy) [inline _common/header.sh]: Same here.

File: _common/header.sh (hunk 1 above)
Comment (trriplejay) [reply _common/header.sh]: updated

File: _common/header.sh (hunk 2 above)
Comment (trriplejay) [reply _common/header.sh]: fixed

**Pull request merged**